### PR TITLE
Adding CUDA SDK 11.4.0

### PR DESCRIPTION
**Pull request by fspiga**
CUDA SDK 11.4.0 Release Notes: https://docs.nvidia.com/cuda/cuda-toolkit-release-notes/index.html

### CODE REVIEW

**Comment by fspiga** [comment]
I am testing locally and I get the following:
```
fspiga@dev-box:spack [cuda-11.4.0]$ spack mirror create -d ~/spack-mirror -D cuda@11.4.0
==> Adding package cuda@11.4.0 to mirror
==> Fetching https://developer.download.nvidia.com/compute/cuda/11.4.0/local_installers/cuda_11.4.0_470.42.01_linux.run
==> Fetching https://developer.download.nvidia.com/compute/cuda/11.4.0/local_installers/cuda_11.4.0_470.42.01_linux.run
==> Fetching https://developer.download.nvidia.com/compute/cuda/11.4.0/local_installers/cuda_11.4.0_470.42.01_linux.run
==> Warning: Error while fetching cuda@11.4.0
  signed integer is greater than maximum
==> Adding package libiconv@1.16 to mirror
==> Adding package libxml2@2.9.10 to mirror
==> Adding package pkgconf@1.7.4 to mirror
==> Adding package xz@5.2.5 to mirror
==> Adding package zlib@1.2.11 to mirror
==> Successfully updated mirror in file:///home/fspiga/spack-mirror
  Archive stats:
    5    already present
    0    added
    1    failed to fetch.
==> Error: Failed downloads:
cuda@11.4.0
```

The bin files exceed 2 GBytes...
```
fspiga@dev-box:tmp $ ls -altr
total 9550427
-rw-rw----  1 fspiga fspiga 3773273383 Jun 23 19:42 cuda_11.4.0_470.42.01_linux.run
-rw-rw----  1 fspiga fspiga 3054682826 Jun 23 19:46 cuda_11.4.0_470.42.01_linux_ppc64le.run
-rw-rw----  1 fspiga fspiga 2944850156 Jun 23 19:52 cuda_11.4.0_470.42.01_linux_sbsa.run
```

**Comment by Rombur** [comment]
The PR looks good but it seems that you hit a problem similar to the one that was reported in https://github.com/spack/spack/issues/24589

File: var/spack/repos/builtin/packages/cuda/package.py [hunk 1]
```diff
@@ -23,6 +24,10 @@
 #    format returned by platform.system() and 'arch' by platform.machine()
 
 _versions = {
+    '11.4.0': {
```
Comment (ax3l) [inline var/spack/repos/builtin/packages/cuda/package.py:28]: Thanks!

Any supported compiler changes that we should put in `CudaPackage`?

Comment (fspiga) [reply var/spack/repos/builtin/packages/cuda/package.py:28]: Not for `x86_64` and `ppc64le`.

However I see no entries in `lib/spack/spack/build_systems/cuda.py` about `aarch64`. I'll consult colleagues before open an additional PR for those (and I will do some testing as well eventually).

Comment (fspiga) [reply var/spack/repos/builtin/packages/cuda/package.py:28]: I am going to kick-off a conversation internally about this point, we all been a bit distracted by many virtual conferences in the past 3 weeks. I'd suggest to progress with the merge now while we make a proper assessment.

Comment (ax3l) [reply var/spack/repos/builtin/packages/cuda/package.py:28]: @fspiga no worries, send a PR when you know more in case something did change after all.

**Review by ax3l — APPROVED** [review]

**Pull request merged**